internal/receiver/discoveryreceiver/statement_evaluator_test.go

```diff
@@ -66,31 +64,26 @@ func TestStatementEvaluation(t *testing.T) {
 							}
 							require.NoError(t, cfg.Validate())
 
-							plogs := make(chan plog.Logs)
-
 							// If debugging tests, replace the Nop Logger with a test instance to see
 							// all statements. Not in regular use to avoid spamming output.
 							// logger := zaptest.NewLogger(t)
 							logger := zap.NewNop()
 							cStore := newCorrelationStore(logger, time.Hour)
+
+							// Change the emit channel to a buffered channel to make the test synchronous
```

Comment:
Wouldn't it be a slightly more realistic test to run the statement evaluator in its own goroutine, and then not override the channel with a buffered channel and just block until it receives?

Reply:
Maybe, but why?

Reply:
I can apply this if you want

Reply:
Made it async